.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,161 @@
+name: Build & Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-18.04
+
+    steps:
+    - name: Set env
+      run: |
+          echo "::set-env name=KONG_SOURCE_LOCATION::$GITHUB_WORKSPACE"
+          echo "::set-env name=KONG_BUILD_TOOLS_LOCATION::$HOME/kong-build-tools"
+        
+    - uses: azure/docker-login@v1
+      with:
+        username: ${{ secrets.KONG_USERNAME }}
+        password: ${{ secrets.KONG_PASSWORD }}
+
+    - name: Checkout Kong source code
+      uses: actions/checkout@v2
+
+    - name: Setup Kong Build Tools
+      run: make setup-kong-build-tools
+
+    - name: Build Test Image
+      run: pushd $KONG_BUILD_TOOLS_LOCATION && make kong-test-container && popd
+
+  lint-unit-tests:
+    name: Lint & Unit
+    runs-on: ubuntu-18.04
+    needs: build
+
+    strategy:
+      matrix:
+        suite: [unit, lint]
```

Comment:
This does not need to be split further, Lint and Unit should be executed sequentially inside the same job, like Travis and the old PR. Because lint finishes extremely quickly, running it concurrently with `unit` actually slows things down.